File: sdks/csharp/src/NovaBilling/AddOns/Requests/CreateAddOnDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CreateAddOnDto
{
    /// <summary>
    /// Display name
    /// </summary>
    [JsonPropertyName("name")]
    public required string Name { get; set; }

    /// <summary>
    /// Unique code for the add-on
    /// </summary>
    [JsonPropertyName("code")]
    public required string Code { get; set; }

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    /// <summary>
    /// Custom name shown on invoices
    /// </summary>
    [JsonPropertyName("invoiceDisplayName")]
    public string? InvoiceDisplayName { get; set; }

    /// <summary>
    /// Prices in different currencies
    /// </summary>
    [JsonPropertyName("prices")]
    public IEnumerable<AddOnPriceDto> Prices { get; set; } = new List<AddOnPriceDto>();

    /// <summary>
    /// Backdate createdAt (ISO 8601). For data imports.
    /// </summary>
    [JsonPropertyName("createdAt")]
    public string? CreatedAt { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/AddOns/Requests/CreateAddOnDto.php
<?php

namespace NovaBilling\AddOns\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Types\AddOnPriceDto;
use NovaBilling\Core\Types\ArrayType;

class CreateAddOnDto extends JsonSerializableType
{
    /**
     * @var string $name Display name
     */
    #[JsonProperty('name')]
    public string $name;

    /**
     * @var string $code Unique code for the add-on
     */
    #[JsonProperty('code')]
    public string $code;

    /**
     * @var ?string $description
     */
    #[JsonProperty('description')]
    public ?string $description;

    /**
     * @var ?string $invoiceDisplayName Custom name shown on invoices
     */
    #[JsonProperty('invoiceDisplayName')]
    public ?string $invoiceDisplayName;

    /**
     * @var array<AddOnPriceDto> $prices Prices in different currencies
     */
    #[JsonProperty('prices'), ArrayType([AddOnPriceDto::class])]
    public array $prices;

    /**
     * @var ?string $createdAt Backdate createdAt (ISO 8601). For data imports.
     */
    #[JsonProperty('createdAt')]
    public ?string $createdAt;

    /**
     * @param array{
     *   name: string,
     *   code: string,
     *   prices: array<AddOnPriceDto>,
     *   description?: ?string,
     *   invoiceDisplayName?: ?string,
     *   createdAt?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->name = $values['name'];
        $this->code = $values['code'];
        $this->description = $values['description'] ?? null;
        $this->invoiceDisplayName = $values['invoiceDisplayName'] ?? null;
        $this->prices = $values['prices'];
        $this->createdAt = $values['createdAt'] ?? null;
    }
}
